File: examples/python/buggy.py
"""A deliberately buggy example program for sl-dbg integration tests.

Run with sl-dbg:
    sl-dbg start --lang python --program examples/python/buggy.py --stop-on-entry

Try setting a conditional breakpoint:
    sl-dbg break examples/python/buggy.py:18 --if "item < 0"
"""

def compute(item: int) -> int:
    # Bug: silently misbehaves on negative inputs.
    return 1000 // item if item else 0


def process(items):
    total = 0
    for item in items:                # line 17
        x = compute(item)             # line 18
        total += x                    # line 19
    return total


if __name__ == "__main__":
    data = [10, 5, 2, 0, -1, 4]       # line 25
    result = process(data)            # line 26
    print(f"result={result}")         # line 27

File: examples/go/buggy.go
// A deliberately buggy example program for sl-dbg integration tests.
//
// Run with sl-dbg:
//
//	sl-dbg start --lang go --program ./examples/go/buggy.go --stop-on-entry
//	sl-dbg break ./examples/go/buggy.go:22 --if "item < 0"
//	sl-dbg continue
//	sl-dbg locals
package main

import "fmt"

func compute(item int) int {
	// Bug: panics on item == 0; silently negative on item < 0.
	if item == 0 {
		return 0
	}
	return 1000 / item
}

func process(items []int) int {
	total := 0
	for _, item := range items { // line 22
		x := compute(item)   // line 23
		total += x           // line 24
	}
	return total
}

func main() {
	data := []int{10, 5, 2, 0, -1, 4}
	result := process(data)
	fmt.Printf("result=%d\n", result)
}
